hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -262,109 +246,56 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
-			Kind:     "ClusterRole",
-			Name:     "edit",
-		}
-		mcsRoleBinding.Subjects = []rbacv1.Subject{
-			{
-				Kind:      "ServiceAccount",
-				Name:      mcsServiceAccount.Name,
-				Namespace: mcsServiceAccount.Namespace,
-			},
+		userDataSecret := IgnitionUserDataSecret(hcluster.Namespace, hcluster.Name, nodePool.Status.Version)
+		if err := r.Delete(ctx, userDataSecret); err != nil && !apierrors.IsNotFound(err) {
+			return reconcile.Result{}, fmt.Errorf("failed to delete token Secret: %w", err)
 		}
-		return nil
-	}); err != nil {
-		return ctrl.Result{}, err
-	} else {
-		span.AddEvent("reconciled mcs role binding", trace.WithAttributes(attribute.String("result", string(result))))
 	}
 
-	r.Log.Info("Reconciling MCS Deployment")
-	if result, err := controllerutil.CreateOrUpdate(ctx, r.Client, mcsDeployment, func() error {
-		return reconcileMCSDeployment(nodePool, mcsDeployment, mcsServiceAccount, releaseImage)
-	}); err != nil {
-		meta.SetStatusCondition(&nodePool.Status.Conditions, metav1.Condition{
-			Type:               string(hyperv1.IgnitionEndpointAvailable),
-			Status:             metav1.ConditionFalse,
-			ObservedGeneration: nodePool.Generation,
-			Reason:             hyperv1.MachineConfigServerDeploymentUnavailableReason,
-			Message:            err.Error(),
-		})
-		return ctrl.Result{}, err
-	} else {
-		newCondition := metav1.Condition{
-			Type:   string(hyperv1.IgnitionEndpointAvailable),
-			Status: metav1.ConditionFalse,
-			Reason: hyperv1.MachineConfigServerDeploymentUnavailableReason,
-		}
-		for _, cond := range mcsDeployment.Status.Conditions {
-			if cond.Type == appsv1.DeploymentAvailable && cond.Status == corev1.ConditionTrue {
-				newCondition = metav1.Condition{
-					Type:   string(hyperv1.IgnitionEndpointAvailable),
-					Status: metav1.ConditionTrue,
-					Reason: hyperv1.MachineConfigServerDeploymentAsExpected,
-				}
-				break
-			}
-		}
-		newCondition.ObservedGeneration = nodePool.Generation
-		meta.SetStatusCondition(&hcluster.Status.Conditions, newCondition)
-		r.Log.Info("reconciled MCS deployment", "result", result)
-		span.AddEvent("reconciled mcs deployment", trace.WithAttributes(attribute.String("result", string(result))))
-	}
+	controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name).Name
+	tokenSecret := TokenSecret(hcluster.Namespace, hcluster.Name, targetVersion)
+	userDataSecret := IgnitionUserDataSecret(controlPlaneNamespace, nodePool.GetName(), targetVersion)
 
-	r.Log.Info("Reconciling MCS Service")
-	if result, err := controllerutil.CreateOrUpdate(ctx, r.Client, mcsService, func() error {
-		mcsService.Spec.Ports = []corev1.ServicePort{
-			{
-				Name:       "http",
-				Protocol:   corev1.ProtocolTCP,
-				Port:       80,
-				TargetPort: intstr.FromInt(8080),
-			},
+	r.Log.Info("Reconciling token Secret")
+	if result, err := controllerutil.CreateOrUpdate(ctx, r.Client, tokenSecret, func() error {
```

Comment:
Set an owner reference and then you don't need to explicitly delete this secret during the NodePool deletion flow

Reply:
who would be the owner since the NodePool live in a different ns?

Reply:
You're right, nevermind....